cmd/cmd.go

```diff
@@ -79,18 +79,34 @@ var rootCmd = &cobra.Command{
 }
 
 func directoryHandler(dirname string) rftp.FileHandler {
-	fi, err := ioutil.ReadDir(dirname)
-	if err != nil {
-		return nil
+	type file struct {
+		path string
+		info os.FileInfo
 	}
+
+	var files []file
+	filepath.Walk(dirname, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			log.Print(err)
+			return nil
+		}
+
+		if !info.IsDir() {
+			p := path[len(dirname)+1:] // relative to the served directory
```

Comment:
This doesn't work for me. If I use `.` as path, it cuts too much of a path and the served files contain only broken paths. An alternative that works for me is:
```
p, err := filepath.Rel(dirname, path)
```